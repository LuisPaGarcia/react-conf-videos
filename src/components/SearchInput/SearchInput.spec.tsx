import * as React from 'react'
import { SearchInputInner } from './SearchInput'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('SearchInputInner', () => {
  const mockFN = jest.fn()

  const shallowComp = () => {
    return shallow<any>(
      <SearchInputInner
        filterValue=''
        onInputChange={mockFN}
        onKeyUpHandlers={mockFN}
        setRef={mockFN}
        isActive={true}
        myRef={null}
        blurRef={mockFN}
      />
    )
  }

  it('should render', () => {
    const comp = shallowComp()
    expect(toJson(comp)).toMatchSnapshot()
  })

  it('should call onKeyUpHandlers on onKeyUp with keyCode 13', () => {
    const comp = shallowComp()
    comp.find('.root').simulate('keyUp', {keyCode: 13})
    expect(mockFN).toHaveBeenCalled()
  })

})