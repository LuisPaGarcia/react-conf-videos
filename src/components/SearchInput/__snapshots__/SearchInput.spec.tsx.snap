// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should dispatch correct actions on change handler call 1`] = `
<Connect(pure(withStateHandlers(withHandlers(Component))))>
  <pure(withStateHandlers(withHandlers(Component)))
    filter={[Function]}
    filterValue=""
    isActive={false}
    setIsActive={[Function]}
  >
    <withStateHandlers(withHandlers(Component))
      filter={[Function]}
      filterValue=""
      isActive={false}
      setIsActive={[Function]}
    >
      <withHandlers(Component)
        blurRef={[Function]}
        filter={[Function]}
        filterValue=""
        isActive={false}
        myRef={
          <input
            class="root"
            placeholder="Search"
            type="text"
            value=""
          />
        }
        setIsActive={[Function]}
        setRef={[Function]}
      >
        <Component
          blurRef={[Function]}
          filter={[Function]}
          filterValue=""
          isActive={false}
          myRef={
            <input
              class="root"
              placeholder="Search"
              type="text"
              value=""
            />
          }
          onInputChange={[Function]}
          onKeyUpHandlers={[Function]}
          setIsActive={[Function]}
          setRef={[Function]}
        >
          <input
            className="root"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </Component>
      </withHandlers(Component)>
    </withStateHandlers(withHandlers(Component))>
  </pure(withStateHandlers(withHandlers(Component)))>
</Connect(pure(withStateHandlers(withHandlers(Component))))>
`;

exports[`SearchInput should render and contain connected props 1`] = `
<pure(withStateHandlers(withHandlers(Component)))
  filter={[Function]}
  filterValue=""
  isActive={false}
  setIsActive={[Function]}
/>
`;

exports[`SearchInputInner should render 1`] = `
<input
  className="root"
  onChange={[MockFunction]}
  onKeyUp={[MockFunction]}
  placeholder="Search"
  type="text"
  value=""
/>
`;
